### PR TITLE
User signup and authentication with custom user model

### PR DESCRIPTION
**Pull request by katjad**
Also moved book functionality into a separate app books

### CODE REVIEW

File: books/views.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        if form.is_valid():
+            book = form.save(commit=False)
+            book.published_date = timezone.now()
+            book.save()
+            return redirect('book_list')                    
+    else: 
+        form = PostBookForm()
+    return render(request, 'books/book_edit.html', {'form': form})
+
+def signup(request):
```
Comment (Geekfish) [inline books/views.py]: Should this view maybe move into the users app?

File: users/models.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from django.db import models
+from django.contrib.auth.models import AbstractUser, UserManager, Group, Permission
+# all custom user functionality from this tut https://wsvincent.com/django-allauth-tutorial-custom-user-model/
```
Comment (Geekfish) [inline users/models.py:3]: 👍

File: bookx/migrations/0001_initial.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Generated by Django 2.1.2 on 2018-10-03 21:52
```
Comment (Geekfish) [inline bookx/migrations/0001_initial.py]: I believe you can get rid of the migrations in `bookx` now?

**Review by Geekfish — APPROVED** [review]
All seems to be going to a good direction. 
I see that you made 2 apps, books and users. This is a logical separation 👍 

Some thoughts in terms of project structure:
You still have the `bookx` module with your wsgi file and url that link everything together. I would suggest removing app-related things from that directory, so it becomes your "project" directory, which is just the "glue" that brings everything together. So in `bookx` you would only have `wsgi`, `urls` and `settings`.

You can also choose to move `users` and `books` into `bookx`, so in the settings you would have `bookx.users` and `bookx.books`, which offers some "namespacing". This isn't necessary, it's just a convention that you can choose to follow (plenty of projects don't).

File: books/views.py (hunk 1 above)
Comment (katjad) [reply books/views.py]: That was from a previous try and is now redundant, accidentally left in in there!  Will remove now :)

**Review by katjad — COMMENTED** [review]
Thanks for your comments! Good to remove things from bookx. Think I'll leave books and users at same level as bookx for now, as have seen it in a few tutorials, and got used to it now :)

**Comment by katjad** [comment]
So I've added one more commit, removing things as suggested by you, but I'll leave it to you to merge :)

**Comment by Geekfish** [comment]
This is associated to #2

**Pull request merged**